### PR TITLE
Add openssl isolated keys

### PR DESCRIPTION
**Pull request by mnaza**

### CODE REVIEW

**Comment by secumod** [comment]
Nice!

**Comment by Lagovas** [comment]
@mnaza you canceled build, can you rebuild?

File: src/soter/openssl_spi/crypter/db.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2017 Cossack Labs Limited
+ *
+ * This file is part of Hermes.
+ *
+ * Hermes is free software: you can redistribute it and/or modify
```
Comment (storojs72) [inline src/soter/openssl_spi/crypter/db.h]: It seems that this is THEMIS - not HERMES

File: src/soter/openssl_spi/soter_asym_key.c [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    return 0;
+  }
+  return (*(int32_t*)(key))&0xffffff00;
+}
+
+int32_t soter_key_get_length_id(const uint8_t* key, const size_t key_length){
+  if(!key || key_length<sizeof(soter_container_hdr_t)){
+    return 0;
+  }
+  return (int32_t)(key[3])-0x30;
```
Comment (storojs72) [inline src/soter/openssl_spi/soter_asym_key.c]: Magic number

File: src/soter/openssl_spi/soter_engine_consts.h [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+#define SOTER_ASYM_SIGN_DEFAULT_ALG (SOTER_ASYM_EC|SOTER_ASYM_EC_LENGTH_256)
+
+/*
+ * @defgroup SOTER_SYM_ALGORYTHMS_IDS symmetric encription/decryption  algorithms ids
+ * @brief supported symmetric encryption/decription algorithms definitions
+ * @{
+ */
+
+/** AES in ECB mode with pkcs7 padding */
+#define SOTER_SYM_AES_ECB_PKCS7     0x10010000
```
Comment (storojs72) [inline src/soter/openssl_spi/soter_engine_consts.h]: ECB, XTS? There is no information about these modes in documentation

Comment (gene-eu) [inline src/soter/openssl_spi/soter_engine_consts.h]: I believe we shouldn't have either of those anywhere

**Comment by vixentael** [comment]
hey, team

are we going to merge this into master before 0.9.5?

what is your final decision? @mnaza

/cc @gene-eu, @storojs72

**Comment by vixentael** [comment]
This thing needs careful review :)

**Comment by vixentael** [comment]
We'll merge it to `cossacklabs:keys_process_isolation` branch, and update later.

**Pull request merged**